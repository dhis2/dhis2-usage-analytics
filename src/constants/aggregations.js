import i18n from '@dhis2/d2-i18n'

export const SUM = 'SUM'
export const AVERAGE = 'AVERAGE'

const AGGREGATIONS = {
    [SUM]: { value: SUM, label: i18n.t('Sum') },
    [AVERAGE]: { value: AVERAGE, label: i18n.t('Average') },
}
export default AGGREGATIONS
