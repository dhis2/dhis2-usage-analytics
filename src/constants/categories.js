import i18n from '@dhis2/d2-i18n'

export const FAVORITE_VIEWS = 'FAVORITE_VIEWS'
export const FAVORITES_SAVED = 'FAVORITES_SAVED'
export const USERS = 'USERS'
export const TOP_FAVORITES = 'TOP_FAVORITES'
export const DATA_VALUES = 'DATA_VALUES'

const CATEGORIES = [
    { value: FAVORITE_VIEWS, label: i18n.t('Favorite views') },
    { value: FAVORITES_SAVED, label: i18n.t('Favorites saved') },
    { value: USERS, label: i18n.t('Users') },
    { value: TOP_FAVORITES, label: i18n.t('Top favorites') },
    { value: DATA_VALUES, label: i18n.t('Data values') },
]
export default CATEGORIES
