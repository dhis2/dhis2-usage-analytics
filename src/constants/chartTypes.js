import i18n from '@dhis2/d2-i18n'

export const ALL = 'ALL'
export const TOTAL = 'TOTAL'

const chartTypes = [
    { value: ALL, label: i18n.t('All') },
    { value: TOTAL, label: i18n.t('Total') },
]

export default chartTypes
