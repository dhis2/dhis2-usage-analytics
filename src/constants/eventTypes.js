import i18n from '@dhis2/d2-i18n'

export const CHART_VIEW = 'CHART_VIEW'
export const MAP_VIEW = 'MAP_VIEW'
export const REPORT_TABLE_VIEW = 'REPORT_TABLE_VIEW'
export const EVENT_REPORT_VIEW = 'EVENT_REPORT_VIEW'
export const EVENT_CHART_VIEW = 'EVENT_CHART_VIEW'
export const DASHBOARD_VIEW = 'DASHBOARD_VIEW'
export const DATA_SET_REPORT_VIEW = 'DATA_SET_REPORT_VIEW'

const EVENT_TYPES = {
    [CHART_VIEW]: { value: CHART_VIEW, label: i18n.t('Chart') },
    [MAP_VIEW]: { value: MAP_VIEW, label: i18n.t('Map') },
    [REPORT_TABLE_VIEW]: {
        value: REPORT_TABLE_VIEW,
        label: i18n.t('Report table'),
    },
    [EVENT_REPORT_VIEW]: {
        value: EVENT_REPORT_VIEW,
        label: i18n.t('Event report'),
    },
    [EVENT_CHART_VIEW]: {
        value: EVENT_CHART_VIEW,
        label: i18n.t('Event chart'),
    },
    [DASHBOARD_VIEW]: { value: DASHBOARD_VIEW, label: i18n.t('Dashboard') },
    [DATA_SET_REPORT_VIEW]: {
        value: DATA_SET_REPORT_VIEW,
        label: i18n.t('Data set report'),
    },
}

export default EVENT_TYPES
