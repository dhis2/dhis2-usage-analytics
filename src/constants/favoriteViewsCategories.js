import i18n from '@dhis2/d2-i18n'

export const SUM_ALL = 'SUM_ALL'
export const SUM_TOTAL = 'SUM_TOTAL'
export const AVEGAGE_ALL = 'AVEGAGE_ALL'
export const AVERAGE_TOTAL = 'AVERAGE_TOTAL'

const FAVORITE_VIEWS_CATEGORIES = [
    { id: SUM_ALL, displayName: i18n.t('All favorite views') },
    { id: SUM_TOTAL, displayName: i18n.t('Total views') },
    {
        id: AVEGAGE_ALL,
        displayName: i18n.t('All favorite average views per user'),
    },
    { id: AVERAGE_TOTAL, displayName: i18n.t('Total views per user') },
]
export default FAVORITE_VIEWS_CATEGORIES
