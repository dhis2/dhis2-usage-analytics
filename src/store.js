import { createStore, applyMiddleware, combineReducers } from 'redux'
import ReduxThunk from 'redux-thunk'
import logger from 'redux-logger'
import appStatus from './reducers/appStatus'
import locale from './reducers/locale'
import usageData from './reducers/usageData'
import filter from './reducers/filter'

const middlewares = [ReduxThunk]

const shouldLog = false

if (process.env.NODE_ENV === 'development' && shouldLog) {
    middlewares.push(logger)
}

const rootReducer = combineReducers({
    appStatus,
    locale,
    usageData,
    filter,
})

export default createStore(rootReducer, applyMiddleware(...middlewares))
