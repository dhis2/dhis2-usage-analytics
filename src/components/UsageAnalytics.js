import React, { Component } from 'react'
import PropTypes from '@dhis2/prop-types'
import { connect } from 'react-redux'
import i18n from '@dhis2/d2-i18n'
import { CircularLoader } from '@dhis2/ui-core'
import { LOADING, ERROR } from '../constants/statuses'
import { initApp } from '../actions'
import Error from './Error'
import FilterSideBar from './FilterSideBar'
import Chart from './Chart'
import Table from './Table'

export class UsageAnalytics extends Component {
    componentDidMount() {
        this.props.initApp()
    }

    render() {
        const { appStatus } = this.props

        if (appStatus === LOADING) {
            return <CircularLoader overlay size="large" />
        }

        if (appStatus === ERROR) {
            return <Error message={i18n.t('Error loading application.')} />
        }

        return (
            <main className="uaa-container">
                <FilterSideBar />
                <section className="uaa-content">
                    <Chart />
                    <Table />
                </section>
            </main>
        )
    }
}

UsageAnalytics.propTypes = {
    appStatus: PropTypes.string,
    initApp: PropTypes.func,
}

const mapStateToProps = ({ appStatus }) => ({
    appStatus,
})

export default connect(mapStateToProps, {
    initApp,
})(UsageAnalytics)
