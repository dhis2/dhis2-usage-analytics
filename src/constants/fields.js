import * as CATS from './categories'
import * as FAV_VIEW_CATS from './favoriteViewsCategories'
import * as PROPS from './properties'

const FAVORITE_VIEWS_TOTAL_FIELDS = [
    PROPS.date,
    PROPS.mapViews,
    PROPS.chartViews,
    PROPS.pivotTableViews,
    PROPS.eventReportViews,
    PROPS.eventChartViews,
    PROPS.dashboardViews,
    PROPS.dataSetReportViews,
    PROPS.totalViews,
]

const FAVORITE_VIEWS_AVERAGE_FIELDS = [
    PROPS.date,
    PROPS.averageMapViews,
    PROPS.averageChartViews,
    PROPS.averagePivotTableViews,
    PROPS.averageEventReportViews,
    PROPS.averageEventChartViews,
    PROPS.averageDashboardViews,
    PROPS.averageViews,
]

export const FIELDS = {
    [CATS.FAVORITE_VIEWS]: {
        [FAV_VIEW_CATS.SUM_ALL]: FAVORITE_VIEWS_TOTAL_FIELDS,
        [FAV_VIEW_CATS.SUM_TOTAL]: FAVORITE_VIEWS_TOTAL_FIELDS,
        [FAV_VIEW_CATS.AVERAGE_ALL]: FAVORITE_VIEWS_AVERAGE_FIELDS,
        [FAV_VIEW_CATS.AVERAGE_TOTAL]: FAVORITE_VIEWS_AVERAGE_FIELDS,
    },
    [CATS.FAVORITES_SAVED]: [
        PROPS.date,
        PROPS.savedMaps,
        PROPS.savedCharts,
        PROPS.savedPivotTables,
        PROPS.savedEventReports,
        PROPS.savedEventCharts,
        PROPS.savedDashboards,
        PROPS.savedIndicator,
    ],
    [CATS.USERS]: [PROPS.date, PROPS.activeUsers, PROPS.users],
    [CATS.TOP_FAVORITES]: [
        PROPS.position,
        PROPS.name,
        PROPS.views,
        PROPS.id,
        PROPS.created,
    ],
    [CATS.DATA_VALUES]: [PROPS.date, PROPS.savedDataValues],
}
export default FIELDS
