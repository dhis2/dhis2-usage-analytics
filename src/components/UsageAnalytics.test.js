import React from 'react'
import { mount } from 'enzyme'
import createSerializer from '../../test/serializer'
import store from '../store'
import { Provider } from 'react-redux'
import { UsageAnalytics } from './UsageAnalytics'
import { LOADING, READY, ERROR } from '../constants/statuses'

expect.addSnapshotSerializer(createSerializer(3))

const defaultProps = {
    initApp: jest.fn(),
    appStatus: READY,
}

describe('<UsageAnalytics/>', () => {
    it('Matches the snapshot', () => {
        const options = {
            wrappingComponent: Provider,
            wrappingComponentProps: { store },
        }
        const tree = mount(<UsageAnalytics {...defaultProps} />, options)
        expect(tree).toMatchSnapshot()
    })
    it('Renders a <CircularProgress/> when appStatus equals LOADING', () => {
        const options = {
            wrappingComponent: Provider,
            wrappingComponentProps: { store },
        }
        const props = { ...defaultProps, appStatus: LOADING }
        const wrapper = mount(<UsageAnalytics {...props} />, options)
        expect(wrapper.find('CircularProgress').length).toEqual(1)
    })

    it('Renders an <Error/> when appStatus equals ERROR', () => {
        const options = {
            wrappingComponent: Provider,
            wrappingComponentProps: { store },
        }
        const props = { ...defaultProps, appStatus: ERROR }
        const wrapper = mount(<UsageAnalytics {...props} />, options)
        expect(wrapper.find('Error').length).toEqual(1)
    })
})
