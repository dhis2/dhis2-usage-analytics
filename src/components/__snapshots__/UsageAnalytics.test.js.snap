// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UsageAnalytics/> Matches the snapshot 1`] = `
<main>
  <ConnectFunction>
    <FilterSideBar>
      <aside />
    </FilterSideBar>
  </ConnectFunction>
  <section>
    <ConnectFunction>
      <Chart />
    </ConnectFunction>
    <ConnectFunction>
      <Table />
    </ConnectFunction>
  </section>
</main>
`;
